### PR TITLE
chore(dsl): Forbid global auth

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Override any global auth config

Depends on https://github.com/island-is/island.is/pull/16405 and https://github.com/island-is/island.is/pull/16375

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208509734870149

## Summary by CodeRabbit

- **New Features**
	- Enabled global authentication for multiple services, enhancing security.
	- Updated ingress configurations to reflect new global authentication settings.

- **Bug Fixes**
	- Removed unnecessary demo ingress configurations, simplifying service setup.

- **Documentation**
	- Improved clarity in type definitions and interfaces related to ingress configurations.

- **Chores**
	- Adjusted resource limits and health check paths for better resource management across services.

### CODE REVIEW

File: infra/src/dsl/dsl.ts [hunk 1]
```diff
@@ -459,10 +460,31 @@
...
+    )) {
+      for (const annotation of Object.keys(annotations)) {
+        if (!(annotation in FORCED_ANNOTATIONS)) {
+          continue
+        }
+        // Apply any forced annotations
+        annotations[annotation] =
+          FORCED_ANNOTATIONS[annotation] ?? annotations[annotation]
+      }
+    }
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply infra/src/dsl/dsl.ts]: This is the intention. I changed the variable names and comment to make it clearer what the intent is.

[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply infra/src/dsl/dsl.ts]: This is the intention. I changed the variable names and comment to make it clearer what the intent is.

File: charts/identity-server/values.prod.yaml [hunk 2]
```diff
@@ -787,7 +779,7 @@ services-auth-public-api:
     primary-alb:
       annotations:
         kubernetes.io/ingress.class: 'nginx-external-alb'
-        nginx.ingress.kubernetes.io/enable-global-auth: 'false'
+        nginx.ingress.kubernetes.io/enable-global-auth: 'true'
```
Comment (busla) [inline charts/identity-server/values.prod.yaml:782]: why is global auth now enabled on prod?

**Review by busla — CHANGES_REQUESTED** [review]

File: charts/identity-server/values.prod.yaml (hunk 2 above)
Comment (AndesKrrrrrrrrrrr) [reply charts/identity-server/values.prod.yaml:782]: It shouldn't be. Working on a fix

File: infra/src/dsl/dsl.ts [hunk 3]
```diff
@@ -459,10 +471,28 @@ export class ServiceBuilder<ServiceType extends string> {
...
+    for (const [env, annotations] of Object.entries(ingressAnnotations)) {
+      for (const annotation of Object.keys(annotations)) {
+        const overrides = OVERRIDE_ANNOTATIONS[env as OpsEnv] ?? {}
+        if (!(annotation in overrides)) {
+          // Leave non-overriddes as is
+          continue
+        }
+        // Apply overrides
+        annotations[annotation] = overrides[annotation]
+      }
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply infra/src/dsl/dsl.ts:494]: No, modifications are intended. This mutates the self object, for chaining, and declarative configuration.

**Comment by AndesKrrrrrrrrrrr** [comment]
Getting too complex if we want to deal with IDS chart changes

**Pull request closed without merging**